### ISSUE TITLE
Feature idea: "modes"

### ISSUE THREAD

**Issue by willruggiano** [tjdevries/subd#4]
I was watching your streamer earlier and had this idea. Here's the scenario:

You want to do a 10 minute AMA. You run some command, which puts subd into some "mode" which saves messages that you deem questions. Then you have something (stream/obs overlay, or a neovim plugin) which displays the "active" question while you answer it or talk about it, then you run a command and the next one pops up.

There's clearly several micro features in there.
- Being able to filter/tag/query messages
- Maybe some sort of "admin" api which you can use to interact with the subd session. Like rpc or json over websocket, etc
- Dybamically transitioning between "modes" which may affect what your filtering/tagging, etc

Anyways, just an idea!

**Comment by tjdevries** [comment]
Yeah, I really like this kind of thing. I definitely wanna do something like "modes" or "activities" or something, where we can start something and chat can start sending in questions w/ commands, and then maybe even vote or something too (like, they get added numbered on the screen and ppl can say "!vote 1")
